### PR TITLE
Coordinates/velocity bugfix

### LINKED ISSUE #6283: Full-space velocity but unit-spherical position doesn't transform properly
After #6219 was merged, I found what looks like a remaining bug - this fails:

```python
c1 = coord.FK5(ra=150*u.deg, dec=-17*u.deg, radial_velocity=83*u.km/u.s,
               pm_ra_cosdec=-41*u.mas/u.yr, pm_dec=16*u.mas/u.yr)
c1.transform_to(coord.Galactic)
```
with
```TypeError: need a base of the correct representation type, <class 'astropy.coordinates.representation.SphericalRepresentation'>, not <class 'astropy.coordinates.representation.UnitSphericalRepresentation'>```

cc @eteq

### PR DESCRIPTION
**Pull request by adrn**
I think this is a change that I had meant to include in #6219 - this relaxes the `_check_base()` method on any differential to only check that the class is compatible with the representation base (i.e. that the differential class is in `rep._compatible_differentials`). This is needed so that, e.g., transformations like this can succeed (`UnitSphericalRepresentation` + `SphericalCosLatDifferential`):
```python
c1 = coord.FK5(ra=150*u.deg, dec=-17*u.deg, radial_velocity=83*u.km/u.s,
               pm_ra_cosdec=-41*u.mas/u.yr, pm_dec=16*u.mas/u.yr)
c1.transform_to(coord.Galactic)
```

Fixes #6283 

cc @eteq @mhvk

### CODE REVIEW

**Comment by adrn** [comment]
There is a simple test in `test_transformations.py`, but I'm adding a more complete set of checks now to the frame-level transformations.

File: astropy/coordinates/representation.py [hunk 1]
```diff
@@ -1990,10 +1987,10 @@ class BaseDifferential(BaseRepresentationOrDifferential):
 
     @classmethod
     def _check_base(cls, base):
-        if not isinstance(base, cls.base_representation):
-            raise TypeError('need a base of the correct representation type, '
-                            '{0}, not {1}.'.format(cls.base_representation,
-                                                   type(base)))
+        if cls not in base._compatible_differentials:
```
Comment (mhvk) [inline astropy/coordinates/representation.py]: Looking at this again, it is slightly odd to me that `base` has to know which differentials are compatible; it seems more logical for the differentials to know which works to transform them (after all, they have the logic to do that in `represent_as`, etc.). See #6287 with this and possible other "implementation details" that we might consider fixing.

Comment (adrn) [inline astropy/coordinates/representation.py]: Agreed - thanks for the new issue

**Comment by adrn** [comment]
Tests are passing locally

**Comment by mhvk** [comment]
@adrn - there were two CI failures that are not due to you... See #6288.

**Comment by adrn** [comment]
Oh, got it - thanks.

**Comment by bsipocz** [comment]
The nddata failure is gone now, and the table operation one seems to pop up sometimes, I'm not yet sure what's the issue there but it must be unrelated to this PR.

**Comment by eteq** [comment]
(merging b/c as @mhvk pointed out, the failures are unrelated)

**Pull request merged**